homeassistant/components/dlna_dmr/media_player.py

```diff
@@ -586,10 +593,30 @@ async def async_play_media(
         """Play a piece of media."""
         _LOGGER.debug("Playing media: %s, %s, %s", media_type, media_id, kwargs)
         assert self._device is not None
+
+        didl_metadata: str | None = None
+        title: str = ""
+
+        # If media is media_source, resolve it to url and MIME type, and maybe metadata
+        if media_source.is_media_source_id(media_id):
+            sourced_media = await media_source.async_resolve_media(self.hass, media_id)
+            media_type = sourced_media.mime_type
+            media_id = sourced_media.url
+            _LOGGER.debug("sourced_media is %s", sourced_media)
+            if sourced_metadata := getattr(sourced_media, "didl_metadata", None):
+                didl_metadata = didl_lite.to_xml_string(sourced_metadata).decode(
```

Comment:
do you have an example of how you would end up with metadata object passed? It seems like it's a special type and wouldn't be possible easily by play_media service and would never come from media browser.

Reply:
nvm looks like it's part of the integration with dms integration that is to come.